### ISSUE TITLE
[TODO] Implemented forced replacement of `GeLU` processing with standard OP.

### ISSUE THREAD

**Issue by PINTO0309** [PINTO0309/onnx2tf#465]
### Issue Type

Others

### OS

Linux

### onnx2tf version number

1.15.18

### onnx version number

1.13.1

### onnxruntime version number

1.15.1

### onnxsim (onnx_simplifier) version number

0.4.33

### tensorflow version number

2.13.0

### Download URL for ONNX

[vttrack.onnx.zip](https://github.com/PINTO0309/onnx2tf/files/12448377/vttrack.onnx.zip)

### Parameter Replacement JSON

```json
N/A
```

### Description

1. Personal
2. Implemented forced replacement of `GeLU` processing with standard OP.

https://www.tensorflow.org/api_docs/python/tf/keras/activations/gelu

https://www.tensorflow.org/api_docs/python/tf/nn/gelu

Automatically detect the range where the following OP combination is valid and replace it with the `GeLU` OP.

```
0.5 * x * (1 + erf(x / sqrt(2.0)))

# sqrt(2) = 1.4142135
# a = tf.math.sqrt(2.0)
# a
# <tf.Tensor: shape=(), dtype=float32, numpy=1.4142135>
# a == 1.4142135381698608
# <tf.Tensor: shape=(), dtype=bool, numpy=True>
```

- a. x -> b.`Div (a*1.4142135381698608)` -> c.`Erf (b)` -> d.`Add (c+1.0)` -> e.`Mul (a*d)` -> f.`Mul (f*0.5)`
  ```
  INFO: onnx_op_type: Div onnx_op_name: Div_145
  INFO:  input_name.1: 366 shape: [1, 320, 256] dtype: float32
  INFO:  input_name.2: 277 shape: [] dtype: float32
  INFO:  output_name.1: 368 shape: [1, 320, 256] dtype: float32
  INFO: tf_op_type: divide
  INFO:  input.1.x: name: tf.math.add/Add:0 shape: (1, 320, 256) dtype: <dtype: 'float32'> 
  INFO:  input.2.y: shape: (1, 1, 1) dtype: <dtype: 'float32'> 
  INFO:  output.1.output: name: tf.math.divide/truediv:0 shape: (1, 320, 256) dtype: <dtype: 'float32'> 
  
  INFO: onnx_op_type: Erf onnx_op_name: Erf_146
  INFO:  input_name.1: 368 shape: [1, 320, 256] dtype: float32
  INFO:  output_name.1: 369 shape: [1, 320, 256] dtype: float32
  INFO: tf_op_type: erf
  INFO:  input.1.x: name: tf.math.divide/truediv:0 shape: (1, 320, 256) dtype: <dtype: 'float32'> 
  INFO:  output.1.output: name: tf.math.erf/Erf:0 shape: (1, 320, 256) dtype: <dtype: 'float32'> 
  
  INFO: onnx_op_type: Add onnx_op_name: Add_148
  INFO:  input_name.1: 369 shape: [1, 320, 256] dtype: float32
  INFO:  input_name.2: 280 shape: [] dtype: float32
  INFO:  output_name.1: 371 shape: [1, 320, 256] dtype: float32
  INFO: tf_op_type: add
  INFO:  input.1.x: name: tf.math.erf/Erf:0 shape: (1, 320, 256) dtype: <dtype: 'float32'> 
  INFO:  input.2.y: shape: (1, 1, 1) dtype: <dtype: 'float32'> 
  INFO:  output.1.output: name: tf.math.add_1/Add:0 shape: (1, 320, 256) dtype: <dtype: 'float32'> 
  
  INFO: onnx_op_type: Mul onnx_op_name: Mul_149
  INFO:  input_name.1: 366 shape: [1, 320, 256] dtype: float32
  INFO:  input_name.2: 371 shape: [1, 320, 256] dtype: float32
  INFO:  output_name.1: 372 shape: [1, 320, 256] dtype: float32
  INFO: tf_op_type: multiply
  INFO:  input.1.x: name: tf.math.add/Add:0 shape: (1, 320, 256) dtype: <dtype: 'float32'> 
  INFO:  input.2.y: name: tf.math.add_1/Add:0 shape: (1, 320, 256) dtype: <dtype: 'float32'> 
  INFO:  output.1.output: name: tf.math.multiply_6/Mul:0 shape: (1, 320, 256) dtype: <dtype: 'float32'> 
  
  INFO: onnx_op_type: Mul onnx_op_name: Mul_151
  INFO:  input_name.1: 372 shape: [1, 320, 256] dtype: float32
  INFO:  input_name.2: 283 shape: [] dtype: float32
  INFO:  output_name.1: 374 shape: [1, 320, 256] dtype: float32
  INFO: tf_op_type: multiply
  INFO:  input.1.x: name: tf.math.multiply_6/Mul:0 shape: (1, 320, 256) dtype: <dtype: 'float32'> 
  INFO:  input.2.y: shape: (1, 1, 1) dtype: <dtype: 'float32'> 
  INFO:  output.1.output: name: tf.math.multiply_8/Mul:0 shape: (1, 320, 256) dtype: <dtype: 'float32'> 
  ```

![image](https://github.com/PINTO0309/onnx2tf/assets/33194443/7a480c44-cc78-49b8-94d2-ca36b3d9cb8a)

An additional `--enable_approximate` option should be implemented to allow the process to be replaced by an approximate calculation. This allows for an increase in overall model performance in exchange for a small degradation in accuracy.

```
0.5 * x * (1 + tanh(sqrt(2 / pi) * (x + 0.044715 * x^3)))
```
https://github.com/onnx/onnx/blob/main/docs/Changelog.md#gelu-20

However, as shown below, if 0.5 is pre-multiplied, it is automatically converted to `GeLU`. (The `Transpose` may be ignored.)

|Before (onnx)|After (tflite)|
|:-:|:-:|
|![image](https://github.com/PINTO0309/onnx2tf/assets/33194443/06674887-833d-483b-96bb-9f5aec165a09)|![image](https://github.com/PINTO0309/onnx2tf/assets/33194443/3a55d6f7-d2e0-4b56-9313-01b2e5354603)|

**Comment by PINTO0309** [comment]
![image](https://github.com/PINTO0309/onnx2tf/assets/33194443/755cf787-9150-4572-a458-88354ac295f9)

|onnx|tflite|
|:-:|:-:|
|![image](https://github.com/PINTO0309/onnx2tf/assets/33194443/9b2f7a0d-ff76-4e7f-a7e9-60e87cce26b8)|![image](https://github.com/PINTO0309/onnx2tf/assets/33194443/0944f075-6b0c-462a-a5cb-fe0c986f6f8d)|
